README.md

```diff
@@ -74,39 +44,174 @@ This program was created with the following tech stack and packages:
+from the `audio` and `text` inputs into a series of phonemes. Phonemes represent the smallest units of sound within
+words and help us distinguish when the mouth should change shape.
+
+To map these phonemes, the `audio` file, `text` file, and `recognizer` selection is passed to the **Phoneme processor**,
+which calls [**Rhubarb Lip Sync**](https://github.com/DanielSWolf/rhubarb-lip-sync). Rhubarb returns a JSON
+[output](#rhubarb-output) that outlines every phoneme along with a `start` and `end` time interval tag.
+
+#### Rhubarb Output
+
+```JSON
+"mouthCues": [
+    { "start": 0.00, "end": 0.08, "value": "X" },
+    { "start": 0.08, "end": 0.23, "value": "B" },
+    { "start": 0.23, "end": 0.37, "value": "C" },
+    { "start": 0.37, "end": 0.44, "value": "G" },
+    { "start": 0.44, "end": 1.42, "value": "B" },
+    ... ]
+
+```
+
+Rhubarb uses the `recognizer` input to select the engine to use to during the process of creating a phoneme file. When
+```recognizer: english```,  Rhubarb will use the `pocketSphinx` engine to process the english audio. When
+```recognizer: phonetic```, Rhubarb will use the `Phonetic` recognizer instead. It has been observed that the `Phonetic`
+recognizer performs better.
+
+Once the phoneme contents are created, they are sent to the next process.
+
+## The Animation Factory
+
+The animation factory uses the [phoneme output](#rhubarb-output) from the phoneme factory to logically select frames
+from `speaking-portal-project/images` to generate the 2D animation. Each frame corresponds to exactly one phoneme,
+avatar, blink value, and body position. This frame data is compiled into a [frame data output](#frame-data-output) file.
+
+### Frame Data Output
+
+```JSON
+file ./images/character01/X_eyes2.png
+outpoint 0.08
+file ./images/character01/B_eyes0.png
+outpoint 0.20
+file ./images/character01/C_eyes0.png
+outpoint 0.07
+file ./images/character01/G_eyes0.png
+outpoint 0.07
+file ./images/character01/B_eyes0.png
+outpoint 0.98
+```
+
+The [frame data output](#frame-data-output) includes a path to the image and the duration of the frame for the animation
+video. This file is then sent to FFMPEG, a highly portable library capable of rendering image frames into an mp4
+output. Once the video file is created, it is then returned to the user as a response from the API.
+
+## Example
+
+Below is an example of the Barb avatar speaking, blinking, and changing poses.
+
+<img src = "docs/documentation/Images/AmberAnimated.gif" width="250px" height="300px">
 
 ## Limitations
 
-- The phoneme generation process controlled via Rhubarb can be quite computationally intensive, and is the current bottleneck for performance.
+- The phoneme generation process controlled via Rhubarb can be quite computationally intensive, and is the current
+bottleneck for performance.
 - Blink frequency is chosen using a random probability
```

Comment:
Didn't we decide this isn't really a limitation?

I mean realistically the lil animation can't get dry eyes so it would always need to blink randomly right? I like our probabilistic approach

Reply:
LOL "lil animation can't get dry eyes"
I like our approach, too, and I don't think it's a limitation either. Let's check with the others before deleting it